lbry/extras/daemon/daemon.py

```
@@ -2478,30 +2598,54 @@ async def jsonrpc_claim_search(self, **kwargs):
+            if {'claim_id', 'claim_ids'}.issubset(kwargs):
+                raise ValueError("Only 'claim_id' or 'claim_ids' is allowed, not both.")
+            if kwargs.pop('valid_channel_signature', False):
+                kwargs['signature_valid'] = 1
+            if kwargs.pop('invalid_channel_signature', False):
+                kwargs['signature_valid'] = 0
+            if 'has_no_source' in kwargs:
+                kwargs['has_source'] = not kwargs.pop('has_no_source')
         page_num, page_size = abs(kwargs.pop('page', 1)), min(abs(kwargs.pop('page_size', DEFAULT_PAGE_SIZE)), 50)
+        wallet = self.wallet_manager.get_wallet_or_default(kwargs.pop('wallet_id', None))
         kwargs.update({'offset': page_size * (page_num - 1), 'limit': page_size})
-        txos, blocked, _, total = await self.ledger.claim_search(wallet.accounts, **kwargs)
-        result = {
-            "items": txos,
-            "blocked": blocked,
-            "page": page_num,
-            "page_size": page_size
-        }
-        if not kwargs.pop('no_totals', False):
-            result['total_pages'] = int((total + (page_size - 1)) / page_size)
-            result['total_items'] = total
-        return result
+        try:
+            txos, blocked, _, total = await self.ledger.claim_search(wallet.accounts, **kwargs)
+            result = {
+                "items": txos,
+                "blocked": blocked,
+                "page": page_num,
+                "page_size": page_size
+            }
+            if not kwargs.pop('no_totals', False):
+                result['total_pages'] = int((total + (page_size - 1)) / page_size)
+                result['total_items'] = total
+            return result
+        except Exception as e:
+            if self.ledger.config['use_go_hub']:
+                log.warning("failed, trying again without hub")
+                self.ledger.config['use_go_hub'] = False
+                return await self.jsonrpc_claim_search(**kwargs_old)
```

@jeffreypicard two things

1. we shouldn't modify the config like this. if another config is saved elsewhere, this setting will also be saved
2. if this fallback is disabled, a bunch of tests fail. therefore we think this is hiding test failures by falling back to old working code.

more generally, if tests pass for all the new code, then errors here should be treated like production errors -- the sdk should report the error to the user.